tests/fixtures/simple/_sources/pkg-a/2.3.4.5/meta.toml

```diff
@@ -0,0 +1,2 @@
+timestamp = 2022-03-29T06:19:50+00:00
+url = "file:tarballs/pkg-a-2.3.4.5.tar.gz"
```

Comment:
perhaps we could accept `file:` URIs pointing to directories? and then that is interpreted as meaning "tar the directory"?

Reply:
:thinking: that does make sense because we create the sdist ourselves. I think with this PR foliage will unpack the tarball and re-create the sdist.

I think it's better to do what you suggest.

Reply:
Oh lol, I was indeed suggesting that we tar it on the fly, and then foliage will unpack the tarball and sdist it :laughing: I agree that it would make sense to shortcut it... but then it is the case that you are not testing the tarball logic. 

Perhaps a compromise is:
- `file` urls pointing to directories are not tarred and are just sdisted directly
- We add at least one test with a checked-in tarball to make sure that our tar handling works

Reply:
There are a bit too many changes to implement either of those. I will leave this for another PR.